eth/api.go

```diff
@@ -543,3 +543,42 @@ func (api *PrivateDebugAPI) getModifiedAccounts(startBlock, endBlock *types.Bloc
 	}
 	return dirty, nil
 }
+
+type AccessListResult struct {
+	Accesslist *types.AccessList `json:"accessList"`
+	Error      string            `json:"error,omitempty"`
+	GasUsed    hexutil.Uint64    `json:"gasUsed"`
+}
+
+// CreateAccessList creates a EIP-2930 type AccessList for the given transaction.
+// Its the result of the `debug_createAccessList` RPC call.
+// Reexec and BlockNrOrHash can be specified to create the accessList on top of a certain state.
+// The AccessListResult contains an error if the transaction failed.
+func (api *PublicDebugAPI) CreateAccessList(blockNrOrHash rpc.BlockNumberOrHash, reexec uint64, args *ethapi.SendTxArgs) (*AccessListResult, error) {
+	ctx := context.Background()
+	var block *types.Block
+	if number, ok := blockNrOrHash.Number(); ok {
+		if number == rpc.PendingBlockNumber {
+			block, _ = api.eth.miner.Pending()
+		} else {
+			if number == rpc.LatestBlockNumber {
+				block = api.eth.blockchain.CurrentBlock()
+			} else {
+				block = api.eth.blockchain.GetBlockByNumber(uint64(number))
+			}
+		}
+	} else if hash, ok := blockNrOrHash.Hash(); ok {
+		block = api.eth.blockchain.GetBlockByHash(hash)
+	} else {
+		return nil, errors.New("either block number or block hash must be specified")
```

Comment:
I think we should default to `latest` like all other APIs, it's bad practice to make this call a special snowflake.

Reply:
I think `blockNrOrHash` resolves `latest` by default.. ?